### PR TITLE
Integrate rocBLAS fp16 api into Caffe2

### PR DESCRIPTION
**Pull request by rohithkrn**
This PR integrates rocBLAS half and mixed precision APIs in to Caffe2.

### CODE REVIEW

**Review by bddppq — COMMENTED** [review]
Could this test be enabled as well? https://github.com/pytorch/pytorch/blob/467546da81576137724ca817e602bbf1de02ce86/caffe2/python/operator_test/momentum_sgd_test.py#L144

**Comment by petrex** [comment]
fp16 for momentum_sgd is in our plan.

**Comment by rohithkrn** [comment]
@bddppq `test_fp16momentum_sgd` is currently failing, it needs some debugging.

File: caffe2/python/operator_test/momentum_sgd_test.py [hunk 1]
```diff
@@ -139,9 +138,10 @@ def sparse(grad, m, lr, param, i):
...
             sparse)
 
-    @given(n=st.integers(4, 8), nesterov=st.booleans(), **hu.gcs_gpu_only)
-    @unittest.skipIf(not workspace.has_gpu_support, "No gpu support.")
+    @unittest.skipIf(not workspace.has_gpu_support and not workspace.has_hip_support, "No gpu support.")
+    @given(n=st.integers(4, 8), nesterov=st.booleans(), **hu.gcs)
     def test_fp16momentum_sgd(self, n, nesterov, gc, dc):
+        assume(core.IsGPUDeviceType(gc.device_type))
         gpuvers = workspace.GetDeviceProperties(0)["major"]
         if gpuvers < 6:
```
[the remark this answers is not in the document]
Comment (rohithkrn) [reply caffe2/python/operator_test/momentum_sgd_test.py]: Shall I revert this particular change enabling the test? I will enable enable it once we fix the test.

[the remark this answers is not in the document]
Comment (bddppq) [reply caffe2/python/operator_test/momentum_sgd_test.py]: @rohithkrn I started the land process already, let's keep it as we know the test is skipped anyway :-)

**Pull request closed without merging**